app/test_engine/models/test_run.py

```diff
@@ -120,6 +122,32 @@ async def __run_handle_errors(self) -> None:
             self.current_test_suite = test_suite
             await test_suite.run()
 
+            # Check if mandatory suite failed
+            if (
+                self.test_run_execution.certification_mode
+                and test_suite.mandatory
+                and any(
+                    tc.state != TestStateEnum.PASSED for tc in test_suite.test_cases
+                )
+            ):
+                print("Abort execution")
+                self.__cancel_remaining_tests()
+                self.cancel()
+                await self.__display_mandatory_test_failure_prompt()
+                break
+
+    async def __display_mandatory_test_failure_prompt(self) -> None:
+        prompt = (
+            "At least one of the mandatory test cases have failed during a "
+            "test run execution in certification mode.\nAs a consequence, the remaining"
+            " tests have been cancelled."
```

Comment:
Just a suggestion: 
`At least one of the required test cases failed while running in certification mode. As a consequence, the remaining tests were cancelled.”`

Reply:
I'll keep "mandatory" instead of required because that's the word used in the test collection, but I'll update the rest. Thanks for the suggestion